### PR TITLE
make configure tools available at the dataset level

### LINKED ISSUE #9589: Enable configure option for dataset external tool
Hello Team

We(ICPSR @ University of Michigan) are working on getting an external tool built for Dataverse that enables users to run a metadata health check for the dataset they create. 
This tool allows the user to look at suggestions/recommendations and some point make api calls to dataverse to persist the changes the user accepts.
Basically this is a configure option for a dataset, which I believe is currently not enabled.
Please help us in understanding if this option would be enable in the near future or if it is enabled per use case.

Regards
Raghunath Ravi
Technical Lead
Team TurboCurator
CNS - ICPSR 
University Of Michigan

### PR DESCRIPTION
**Pull request by pdurbin**
**What this PR does / why we need it**:

ICPSR is making a tool called TurboCurator. See #9589. Other members of the community have also expressed interest in creating a configure tool at the dataset level.

**Which issue(s) this PR closes**:

- Closes #9589

**Special notes for your reviewer**:

I left a TODO about some potential code consolidation.

**Suggestions on how to test this**:

Add a configure tool at the dataset level and test it. ExternalToolsIT.java has JSON you can use if you like.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

Yes and I'm using a gear/cog ⚙️ to match configure tools at the file level:

Screen Shot 2023-09-18 at 10 33 28 AM

**Is there a release notes update needed for this change?**:

Yes, included.

**Additional documentation**:

Docs have been updated:

- https://dataverse-guide--9925.org.readthedocs.build/en/9925/admin/external-tools.html
- https://dataverse-guide--9925.org.readthedocs.build/en/9925/api/external-tools.html

### CODE REVIEW

**Comment by pdurbin** [comment]
As I mentioned in standup, now that I fixed the icon in 258023e to match the file-level configure tool, this is ready for review.

**Comment by pdurbin** [comment]
I'm pulling this back into development because I realized I forgot to add the popup that configure tools have at the file level. Here's how it looks for the Data Curation Tool:

Screen Shot 2023-09-20 at 4 37 20 PM

I checked with ICPSR and they said they'd still like the popup, as discussed previously.

**Comment by pdurbin** [comment]
With some [help](https://github.com/IQSS/dataverse/pull/9925/commits/89dfb012458a93ec563e9c81f9458c2919b23d51) from @sekmiller (thanks!), I was able to make progress on the popup. Here's how it looks:

Screen Shot 2023-09-26 at 4 05 51 PM
Screen Shot 2023-09-26 at 4 06 10 PM

File: doc/sphinx-guides/source/admin/external-tools.rst [hunk 1]
```diff
@@ -115,7 +115,7 @@ Dataset level explore tools allow the user to explore all the files in a dataset
 Dataset Level Configure Tools
 +++++++++++++++++++++++++++++
 
-Configure tools at the dataset level are not currently supported.
+Dataset level configure tools can be launched by users who have edit access to the dataset and are found under the "Edit Dataset" menu.
```
Comment (sekmiller) [inline doc/sphinx-guides/source/admin/external-tools.rst:118]: maybe two sentences here. a quick read might make you think that the users are found under the the Edit Dataset menu.

File: src/main/webapp/dataset.xhtml [hunk 2]
```diff
@@ -990,6 +1001,19 @@
                             </button>
                         </div>
                     </p:dialog>
+                    <p:dialog id="configureToolDialog" styleClass="smallPopUp" header="#{DatasetPage.datasetConfigureTool.displayNameLang}" widgetVar="configureToolDialog" modal="true">
+                        <p class="help-block">
+                            <h:outputFormat value="#{DatasetPage.datasetConfigureTool.description}" escape="false"/>
+                        </p>
+                        <div class="button-block">
+                            <p:commandLink styleClass="btn btn-default" action="#{DatasetPage.explore(DatasetPage.datasetConfigureTool)}" oncomplete="PF('configureToolDialog').hide();">
```
Comment (sekmiller) [inline src/main/webapp/dataset.xhtml:1009]: should be calling the configure method on the dataset page

**Review by sekmiller — CHANGES_REQUESTED** [review]
Looks good overall, a couple of minor changes.

File: doc/sphinx-guides/source/admin/external-tools.rst (hunk 1 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/admin/external-tools.rst]: Ha, yes. I agree. Two sentences now: f9fba20

**Comment by pdurbin** [comment]
@sekmiller thanks for catching those! Fixed.

**Review by sekmiller — CHANGES_REQUESTED** [review]
Thanks for the changes - ready for QA

**Review by sekmiller — APPROVED** [review]

**Pull request merged**